File: engine-kernel/src/main/java/com/sos/scheduler/engine/kernel/order/OrderQueue.java
package com.sos.scheduler.engine.kernel.order;

import com.sos.scheduler.engine.cplusplus.runtime.Sister;
import com.sos.scheduler.engine.cplusplus.runtime.SisterType;
import com.sos.scheduler.engine.kernel.cppproxy.Order_queueC;

public class OrderQueue implements Sister {  // Iterable<Order>
    private final Order_queueC cppProxy;

    public OrderQueue(Order_queueC cppProxy) {
        this.cppProxy = cppProxy;
    }

    @Override public final void onCppProxyInvalidated() {}

    public final int size()  {
        return cppProxy.java_order_count();
    }

    public static class Type implements SisterType<OrderQueue, Order_queueC> {
        @Override public final OrderQueue sister(Order_queueC proxy, Sister context) { 
            return new OrderQueue(proxy);
        }
    }
}

File: engine-kernel/src/main/scala/com/sos/scheduler/engine/kernel/order/OrderQueue.scala
package com.sos.scheduler.engine.kernel.order

import com.sos.scheduler.engine.cplusplus.runtime.{Sister, SisterType}
import com.sos.scheduler.engine.kernel.cppproxy.Order_queueC

final class OrderQueue(cppProxy: Order_queueC) extends Sister {
  def onCppProxyInvalidated() {}

  def size = cppProxy.java_order_count
}

object OrderQueue {

  object Type extends SisterType[OrderQueue, Order_queueC] {
    final def sister(proxy: Order_queueC, context: Sister): OrderQueue = new OrderQueue(proxy)
  }
}
